paper-input-container.html

```diff
@@ -366,40 +400,37 @@
     _handleValue: function(inputElement) {
       var value = inputElement[this._propertyForValue] || inputElement.value;
 
-      var valid;
-
       if (this.autoValidate) {
         if (inputElement.validate) {
           valid = inputElement.validate(value);
         } else {
           valid = inputElement.checkValidity();
         }
-      } else {
-        valid = true;
+        this.invalid = !valid;
       }
 
       // type="number" hack needed because this.value is empty until it's valid
-      if (value || inputElement.type === 'number' && !valid) {
+      if (value || (inputElement.type === 'number' && !inputElement.checkValidity())) {
         this._inputHasContent = true;
       } else {
         this._inputHasContent = false;
       }
 
-      // notify add-ons
-      for (var addon, i = 0; addon = this._addons[i]; i++) {
-        // need to set all of these, or call method... thanks input type="number"!
-        addon.inputElement = inputElement;
-        addon.value = value;
-        addon.invalid = !valid;
-      }
+      this._updateAddons({
```

Comment:
Could this maybe be public? It could be useful if you're doing manual `auto-validation` in a custom element (like gold-cc-input)